js/lightbeam.js

```diff
@@ -1,9 +1,33 @@
 async function renderGraph() {
-  const canvas = document.getElementById('canvas');
-  const context = canvas.getContext('2d');
   const websites = await store.getAll();
 
-  viz.draw(context, websites);
+  // reformat data for D3
+  const nodesData = [];
+  const linksData = [];
+  // make first party nodes
+  for (const firstParty in websites) {
+    const node = {};
+    node.hostname = firstParty;
+    node.favicon = websites[firstParty].faviconUrl || '';
+    node.party = 'first';
+    nodesData.push(node);
+    // make third party nodes and links
+    for (const thirdParty in websites[firstParty].thirdPartyRequests) {
+      const node = {};
+      const link = {};
+      link.source = firstParty;
+      link.target = thirdParty;
+      node.hostname = thirdParty;
+      // third parties likely don't have favicons
+      node.favicon = '';
+      node.party = 'third';
+      nodesData.push(node);
```

Comment:
No, it doesn't. I have fixed this with a `thirdParties` array that keeps track of third parties I that already have nodes. Only if a third party is not in that array will a new node be created.